packages/contracts/test/10_unit-testing/11_plugin.ts

```diff
@@ -265,22 +339,118 @@ describe('Admin', function () {
         expect(event.args.actions[0].to).to.equal(dummyActions[0].to);
         expect(event.args.actions[0].value).to.equal(dummyActions[0].value);
         expect(event.args.actions[0].data).to.equal(dummyActions[0].data);
-        // note that failureMap is different than allowFailureMap. See DAO.sol for details
+        // note that failureMap is different than allowFailureMap. See `DAO.sol` for details
         expect(event.args.failureMap).to.equal(0);
       }
 
       {
         const proposalId = 1;
 
-        const tx = await plugin.executeProposal(dummyMetadata, dummyActions, 0);
+        const tx = await newPlugin
+          .connect(alice)
+          .executeProposal(dummyMetadata, dummyActions, 0);
 
-        const event = await findEventTopicLog<ExecutedEvent>(
+        const event = await findEventTopicLog<DAOEvents.ExecutedEvent>(
           tx,
-          DAO__factory.createInterface(),
-          IDAO_EVENTS.EXECUTED
+          dao.interface,
+          dao.interface.getEvent('Executed').name
         );
         expect(event.args.callId).to.equal(proposalIdToBytes32(proposalId));
       }
     });
   });
 });
+
+type FixtureResult = {
+  deployer: SignerWithAddress;
+  alice: SignerWithAddress;
+  bob: SignerWithAddress;
+  initializedPlugin: Admin;
+  uninitializedPlugin: Admin;
+  dao: DAO;
+  dummyActions: DAOStructs.ActionStruct[];
+  dummyMetadata: string;
+};
+
+async function fixture(): Promise<FixtureResult> {
+  const [deployer, alice, bob] = await ethers.getSigners();
+
+  const dummyMetadata = ethers.utils.hexlify(
+    ethers.utils.toUtf8Bytes('0x123456789')
+  );
+  const dao = await createDaoProxy(deployer, dummyMetadata);
+
+  const adminPluginImplementation = await new Admin__factory(deployer).deploy();
+  const adminProxyFactory = await new ProxyFactory__factory(deployer).deploy(
+    adminPluginImplementation.address
+  );
+
+  // Create an initialized plugin clone
+  const adminPluginInitdata =
+    adminPluginImplementation.interface.encodeFunctionData('initialize', [
+      dao.address,
+    ]);
+  const tx1 = await adminProxyFactory.deployMinimalProxy(adminPluginInitdata);
```

Comment:
Perhaps we can use better names than `fixture` `tx1`, `event1` etc

Reply:
Fixed in 803b66d94cd419ec64cc077d8dcc4d9e1b294217

Reply:
Improved in 05465403d2d930e45389802a9d6094bfdd9af511.